master/buildbot/db/logs.py

```diff
@@ -345,6 +345,40 @@ def thdcompressLog(conn):
         saved = yield self.db.pool.do(thdcompressLog)
         defer.returnValue(saved)
 
+    def deleteOldLogChunks(self, older_than_timestamp):
+        def thddeleteOldLogs(conn):
+            model = self.db.model
+            res = conn.execute(sa.select([sa.func.count(model.logchunks.c.content)]))
+            count1 = res.fetchone()[0]
+            res.close()
+
+            # update log types older than timestamps
+            res = conn.execute(
+                model.logs.update()
+                .where(model.logs.c.stepid.in_(
+                    sa.select([model.steps.c.id])
+                    .where(model.steps.c.started_at < older_than_timestamp)))
+                .values(type='d')
+            )
+            res.close()
+
+            # query all logs older than timestamps
+            q = sa.select([model.logs.c.id])
+            q = q.select_from(model.logs.join(model.steps, model.steps.c.id == model.logs.c.stepid))
+            q = q.where(model.steps.c.started_at < older_than_timestamp)
+
+            # delete their logchunks
+            res = conn.execute(
+                model.logchunks.delete()
+                .where(model.logchunks.c.logid.in_(q))
```

Comment:
Do you have an idea on whether this will be slow on the first time when clearing a lot of log chunks? I've never pictured `IN` as being particularly fast. This is executed in a thread so I think it won't block the rest of the process, but I wonder about the database. Checking the Postgres docs and looks to be fine there so I would think MySQL and sqlite will also be fine.